internal/v1/handler_post_compose_test.go

```diff
@@ -1223,11 +1228,124 @@ func TestComposeWithSnapshots(t *testing.T) {
+							Gpgkey:   &[]string{"some-gpg-key"},
+							Id:       repoPayloadId.String(),
+						},
+					},
+				},
+			},
+			composerRequest: composer.ComposeRequest{
+				Distribution: "rhel-94",
+				ImageRequest: &composer.ImageRequest{
+					Architecture: "x86_64",
+					ImageType:    composer.ImageTypesGcp,
+					Repositories: []composer.Repository{
+						{
+							Baseurl:     common.ToPtr("https://packages.cloud.google.com/yum/repos/google-compute-engine-el9-x86_64-stable"),
+							Rhsm:        common.ToPtr(false),
+							Gpgkey:      common.ToPtr(gcpGpg),
+							CheckGpg:    common.ToPtr(true),
+							IgnoreSsl:   nil,
+							Metalink:    nil,
+							Mirrorlist:  nil,
+							PackageSets: nil,
+						},
+						{
+							Baseurl:     common.ToPtr("https://packages.cloud.google.com/yum/repos/cloud-sdk-el9-x86_64"),
+							Rhsm:        common.ToPtr(false),
+							Gpgkey:      common.ToPtr(gcpGpg),
+							CheckGpg:    common.ToPtr(true),
+							IgnoreSsl:   nil,
+							Metalink:    nil,
+							Mirrorlist:  nil,
+							PackageSets: nil,
+						}, {
+							Baseurl:     common.ToPtr("https://content-sources.org/snappy/baseos"),
+							Rhsm:        common.ToPtr(false),
+							Gpgkey:      common.ToPtr(rhelGpg),
+							CheckGpg:    common.ToPtr(true),
+							IgnoreSsl:   nil,
+							Metalink:    nil,
+							Mirrorlist:  nil,
+							PackageSets: nil,
+						},
+						{
+							Baseurl:     common.ToPtr("https://content-sources.org/snappy/appstream"),
+							Rhsm:        common.ToPtr(false),
+							Gpgkey:      common.ToPtr(rhelGpg),
+							CheckGpg:    common.ToPtr(true),
+							IgnoreSsl:   nil,
+							Metalink:    nil,
+							Mirrorlist:  nil,
+							PackageSets: nil,
+						},
+					},
+					UploadOptions: makeUploadOptions(t, composer.GCPUploadOptions{
+						Region:            "",
+						Bucket:            common.ToPtr(""),
+						ShareWithAccounts: common.ToPtr([]string{"user:example@example.com"}),
+					}),
+				},
+				Customizations: &composer.Customizations{
+					PayloadRepositories: &[]composer.Repository{
+						{
+							Baseurl:  common.ToPtr("https://content-sources.org/snappy/payload"),
+							CheckGpg: common.ToPtr(true),
+							Gpgkey:   common.ToPtr("some-gpg-key"),
+							Rhsm:     common.ToPtr(false),
+						},
+					},
+					CustomRepositories: &[]composer.CustomRepository{
+						{
+							Baseurl:  &[]string{"http://snappy-url/snappy/payload"},
+							Name:     common.ToPtr("payload"),
+							CheckGpg: common.ToPtr(true),
+							Enabled:  common.ToPtr(false),
+							Gpgkey:   &[]string{"some-gpg-key"},
+							Id:       repoPayloadId.String(),
+						},
+					},
+				},
+			},
+		},
 	}
 
 	for idx, payload := range payloads {
 		fmt.Printf("TT payload %d\n", idx)
 		respStatusCode, body := tutils.PostResponseBody(t, srv.URL+"/api/image-builder/v1/compose", payload.imageBuilderRequest)
+		if respStatusCode != http.StatusCreated {
+			fmt.Printf("Body: %s\n", body)
```

Comment:
Is this a prank show or something? :-)

Other than that all good.

Reply:
I guess Lukas is looking for more proper logging here, am I correct? :)